cli/module_build.go

```diff
@@ -372,3 +411,63 @@ func jobStatusFromProto(s buildpb.JobStatus) jobStatus {
 		return jobStatusUnspecified
 	}
 }
+
+// envOrValue does os.Getenv if input starts with $, otherwise returns value directly.
+func envOrValue(orig string) string {
+	if len(orig) > 0 && orig[0] == '$' {
+		return os.Getenv(orig[1:])
+	}
+	return orig
+}
+
+// flagToCredentials constructs entity credentials option from apiKeyFlags in cli.Context.
+func flagToCredentials(cCtx *cli.Context) (rpc.DialOption, error) {
+	var keyID, key string
+	// careful: envOrValue resolution must come before length check, or else default env var will always override.
+	pair := envOrValue(cCtx.String(loginFlagKeyIDVal))
+	if len(pair) > 0 {
+		keyID, key, _ = strings.Cut(pair, ":")
+	} else {
+		keyID = cCtx.String(loginFlagKeyID)
+		key = cCtx.String(loginFlagKey)
+		if len(keyID) == 0 || len(key) == 0 {
+			return nil, errors.New("pass either --key-id-val, or both of --key-id, --key")
+		}
+	}
+	return rpc.WithEntityCredentials(keyID, rpc.Credentials{
+		Type:    rpc.CredentialsTypeAPIKey,
+		Payload: key,
+	}), nil
+}
+
+// strippedFqdn removes leading 'https://' prefix + trailing '/' from fqdn in case user copied from 'local link' in app.
+func strippedFqdn(orig string) string {
+	return strings.TrimSuffix(strings.TrimPrefix(orig, "https://"), "/")
+}
+
+// RestartModuleAction restarts a module on a robot.
+func RestartModuleAction(c *cli.Context) error {
+	modName := c.String(moduleFlagName)
+	modID := c.String(moduleFlagID)
+	// note: this is xor
+	// todo: use MutuallyExclusiveFlags for this when urfave/cli 3.x is stable
+	if (len(modName) == 0) == (len(modID) == 0) {
+		return fmt.Errorf("provide exactly one of --%s and --%s", moduleFlagName, moduleFlagID)
+	}
+	dialOpt, err := flagToCredentials(c)
```

Comment:
I think we should be able to use the cached credentials / setup some pipieline with app. I don't think we need to accept creds here

Reply:
Brainstorming: Look at connect code samples page API requests to see if we can mimic that

Reply:
I'm guessing the cloud app RobotService is not a good option? https://github.com/viamrobotics/api/blob/main/proto/viam/app/v1/robot.proto#L13-L25
like we don't want it to be a passthrough for the full functionality of the device

Reply:
ah you mean the keys list on `/machine/xxx/connect/api-keys`? that makes sense to me

are there downsides to this? use wrong key by accident?

Reply:
To my knowledge, all robots are auto-generated an APIKey such that resource_type = robot and resource_id = {robot you want to connect to}.  I think there is no harm in finding that key and using that key to auth to the robot

(the user can delete that key, but we can almost-certainly treat that as an edge case, a few parts of our app complain if there are no api keys for a robot)

Reply:
It seems like significantly better UX than making the user go to app.viam (and login via the same auth that they signed into the cli with, and go manually copy-paste api keys into their terminal history)

Reply:
using GetRobotApiKeys in 6d2b7758

I'm also calling GetRobotPart, which has a Part.Secrets list on it -- is there a way to use the part secret directly?

Reply:
I think part secrets are in a weird socally-deprecated technically-supported state on the fleet teams. If I were you, unless you write e2e tests for this, I'd avoid using the secret directly and would use the API key. But I think the secret would work

Reply:
Up to you